src/entities/follower/follower.service.ts

```diff
@@ -45,34 +50,16 @@ export class FollowerService {
 
   async list(filters?: any, paginationArgs?: PaginationArgs) {
     const { first, after, before } = paginationArgs
-    const limit = first ?? 0
 
-    filters = filters ?? {}
-    const options = {}
+    const { result, hasNextPage } = await getPaginatedResults<Follower>({
+      dbModel: this.dbModel,
+      filters,
+      first,
+      after,
+      before,
+    })
 
-    if (first) {
-      // In order to check if there is a next page, fetch one extra record
-      options['limit'] = first + 1
-    }
-    // 'before' and 'after' are mutually exclusive. Because of this:
-    if (after) {
-      const afterDate = new Date(fromCursorHash(after))
-      filters['createdAt'] = { $gt: afterDate.toISOString() }
-    } else if (before) {
-      const beforeDate = new Date(fromCursorHash(before))
-      filters['createdAt'] = { $lt: beforeDate.toISOString() }
-    }
-    const result = await this.dbModel.find(filters, null, options).exec()
-    let hasNextPage = false // Default behavior for empty result
-    if (result?.length > 0) {
-      hasNextPage = result.length === first + 1
-    }
-
-    // Build PaginatedSchool
-    if (hasNextPage && limit > 0) {
-      result.pop()
-    }
-    return {
+    const paginatedResult = {
       edges: result.map((doc) => {
```

Comment:
@frankmangone if result is null / undefined this may crash?

```suggestion
      edges: result?.map((doc) => {
```